### DISCUSSION TITLE
Actividad 4 - Grupo 4

### DISCUSSION THREAD

**Discussion by MatyRusso22** [frlp-utn-ingsoft/catedra#159] [Show and tell]
https://github.com/MatyRusso22/vetsoft

**Comment by ivanduva** [comment]
Buenas! Les puedo pedir el link a donde lo deployaron?

**Reply by MauroBaroni** [reply]
https://vetsoft-qkop.onrender.com/

**Reply by ivanduva** [reply]
Buenas,
Les vamos a pedir que hagan algunos arreglos y cambios:

- En cuanto a la validación de que el teléfono sea numérico, les pedimos que bloqueen los caracteres no numéricos en el form y que lo testeen con un e2e, ya que el que agregaron (test_shouldnt_be_able_to_create_client_with_no_numeric_phone) no es e2e.
- En cuanto al campo city, en ese PR dejaron un bug que lo borra del form al guardar con error en otros campos. El form debería mantener el valor. Lo mismo pasa al editar, no aparece el valor guardado en el select. Además los tests de unidad e integracion testean que no se modifique la ciudad al agregar un valor erróneo. Modifiquen los tests para que prueben que salta el error correspondiente.
- En cuanto a la validación de que el teléfono empiece con 54, les pedimos que agreguen un test de unidad y uno de integración.
- En cuanto a la validación del email, se permite agregar "@vetsoft.com" sin nada adelante. Arreglen eso. Además en los tests de unidad e integración está el mismo problema que en los de la ciudad, testean que no se modifique el teléfono al agregar un valor erróneo. Modifiquen los tests para que prueben que salta el error correspondiente.
- Luego de hacer todo, saquen una release. La documentación que hicieron de las demás releases está bastante buena, así que haganlo de la misma manera.

Recuerden que los cambios deben ser realizados por las personas que se encargaron de cada issue.

Por otro lado, seguimos viendo problemas en los PRs y las reviews. Además de la casi nula interacción, hay algunas malas prácticas como por ejemplo, varios PRs poco atómicos que solucionan varias cuestiones que no están relacionadas entre sí. Este https://github.com/MatyRusso22/vetsoft/pull/21 agrega lo relativo a la CI/CD junto a lo de la regla D102 de los tests estáticos y este https://github.com/MatyRusso22/vetsoft/pull/26 arregla algo relativo al deploy junto a la validación del teléfono. Esto no solo dificulta la review sino que además dificulta solucionar problemas que traigan esos PR luego de mergearlos. También vimos este revert: https://github.com/MatyRusso22/vetsoft/pull/19 que no terminamos de entender la razón. Con una buena review se evita cualquier inconveniente. Igualmente un revert se hace únicamente en un caso muy grave y como última instancia. No creemos que este sea el caso, se podría haber solucionado de otra manera.
Tengan esto en cuenta en las correcciones.

**La última fecha límite de será el último día de cursada antes de las vacaciones: Lunes 8 de Julio 23:59 hs.**

Saludos

**Reply by MatyRusso22** [reply]
Buenas profe, me surgio una consulta.

Al momento de hacer una correccion, tenemos que crear una version nueva con el formato x.x.1, indicando de que se arreglo un bug? o simplemente no hacemos una nueva release y solo arreglamos el error?

**Reply by ivanduva** [reply]
En este caso, les pedimos que saquen una release luego de hacer todas las correcciones. El número de versión está bien, en la release habría que cambiar el último número porque son bugs.

**Reply by MatyRusso22** [reply]
Buenas profe, ya corregimos los bugs que nos menciono.

Tambien le queria comentar que tuve un bug con mi carpeta local de git, y por este termine pusheando al repositorio de manera no intencionada. Lo solucione realizando una nueva rama para hacer un revert, y luego haciendo un pr para mergear ese revert asi volvia al estado antes de mi error. En ese pr entro mas en detalle sobre el tema (https://github.com/MatyRusso22/vetsoft/pull/36)

Saludos! Equipo 4 (Sivori, Daza, Ruival, Baroni, Russo)

**Reply by ivanduva** [reply]
Buenas,

- No vemos la corrección que les pedimos sobre los tests de unidad e integración sobre el email. Serían `test_update_client_with_error_wrong_email_not_end_vetsoft `y `test_edit_user_with_invalid_data_test_email_not_end_vetsoft` respectivamente.
- En cuanto a los tests de unidad e integración del campo ciudad, son irrelevantes porque el arreglo que hicieron es sobre el form. Los tests de unidad e integración no testean la interfaz. Además no les pedimos que agreguen otros tests sino que modifiquen los que habían agregado antes (`test_update_client_with_error_wrong_city `y `test_edit_user_with_invalid_data_test_city`). Lo que deberían hacer esos tests es testear que la función de validación del modelo devuelva el error correspondiente cuando la ciudad no es correcta (unidad) y que la API devuelva el error correspondiente cuando la ciudad no es correcta (integración).

Ambos casos son parte de lo mismo. Como dijimos antes, están probando que el cliente no se modifique en caso de datos erróneos. Lo cual no es específico ya que el hecho de que no se modifique puede pasar por muchas razones y el test no estaría atajando el error correspondiente. Por eso les pedimos que testeen que salte el error en específico en cada caso.

Otra cuestión menor que vimos es que el número de versión que usaron en este caso no es del todo correcto. Cuando se solucionan bugs sin agregar features, en general se cambia el último número. Para saber más pueden revisar este link que también está en la documentación de la actividad: https://semver.org/lang/es/.

Después de solucionar los problemas, saquen una nueva release. Y recuerden como siempre que cada cosa deberá resolverla el encargado de cada issue.

Saludos

**Reply by MatyRusso22** [reply]
Buenas profe, ahi corregimos lo que nos pidió, e hicimos una nueva release 1.7.1. Saludos!

**Comment by ivanduva** [comment]
Perfecto, queda aprobado el trabajo integrador.

Saludos
